scripts/abi_artifact_generator/templates/functions.inc.jinja_template

```diff
@@ -2,36 +2,44 @@
 This template file is distributed under the MIT License. See LICENSE.md for details.
 #*/
 
-/*( generation_notice )*/
+/*(- generation_notice )*/
 
+#ifndef _MSC_VER
 #include <stdalign.h>
+#endif
+
 #include <stdbool.h>
+#include <stddef.h>
 
 #include "common.h"
 #include "functions.h"
+#include "types.h"
 
 /*
  * Functions designed to test argument passing
  */
 
 /*{ for name, arguments in argument_functions.items() -}*/
-void __attribute__((noinline, weak)) ABIDEF /*( name )*/(
+NOINLINE void ABIDEF test_/*( name )*/(
   /*{- for type in arguments -}*/
   /*( type )*/ argument_/*( loop.index - 1 )*/
   /*{- if not loop.last -}*/ , /*{ endif -}*/
   /*{- endfor -}*/
 ) {
-  puts("    Function: \"/*( name )*/\"");
-  puts("    Arguments:");
-/*{ for type in arguments }*/
-  PRINT(/*( type )*/, &argument_/*( loop.index - 1 )*/);
-/*{- endfor }*/
-
-  puts("    ReturnValue:");
-  puts("");
+  CurrentOffset = 0;
+
+  /*{- for type in arguments }*/
+  if (sizeof(/*( type )*/) <= ARGUMENT_SIZE_LIMIT(argument_/*( loop.index - 1 )*/) && !IS_PACKED(/*( type )*/)) {
+    *(/*( type )*/ *) ((uint8_t *) TemporaryLocation + CurrentOffset) = argument_/*( loop.index - 1 )*/;
+    CurrentOffset += sizeof(/*( type )*/);
+  } else {
+    *(/*( type )*/ **) ((uint8_t *) TemporaryLocation + CurrentOffset) = &argument_/*( loop.index - 1 )*/;
+    CurrentOffset += sizeof(/*( type )*/ *);
+  }
```

Comment:
I cannot read this.

1. Thoroughly describe what's going on here.
2. Can you reduce the amount of `/* ... */` per line by introducing some temporary variables and local `typedefs`? The code is really really hard to read. `typedefs` lifetime is limited to the current scope, so you can "redefine them" multiple times with the same name (in different scopes).

Reply:
Closing all the documentation-related threads for now.